server/app/src/androidTest/java/sh/calaba/instrumentationbackend/StatusReporterObject.java

```diff
@@ -0,0 +1,76 @@
+package sh.calaba.instrumentationbackend;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class StatusReporterObject {
+
+    public static void report(String method,
+                              String message,
+                              StatusReporter.FinishedState extraState,
+                              boolean hasData) {
+
+
+        System.out.println("Failure file: " + getOutputFile(FAILURE_FILE_PATH));
+        System.out.println("Finished file: " + getOutputFile(FINISHED_FILE_PATH));
+
+        if (hasData) {
+
+            System.out.println("method: " + method);
+
+            if ("report-failure".equals(method)) {
+                try {
+                    reportFailure(message);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else if ("report-finished".equals(method)) {
+                try {
+                    reportFinished(extraState);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else {
+                throw new RuntimeException("No such method '" + method + "'");
+            }
+
+        } else {
+            throw new RuntimeException("No data sent");
+        }
+    }
+
+    private static final String FAILURE_FILE_PATH = "calabash_failure.out";
+    private static final String FINISHED_FILE_PATH = "calabash_finished.out";
+
+    private static void reportFailure(String message) throws IOException {
+        System.out.println("Failure state: " + message);
+
+        dumpPublicFile(FAILURE_FILE_PATH, message);
+    }
+
+    private static void reportFinished(StatusReporter.FinishedState finishedState) throws IOException {
+        System.out.println("Finished state: " + finishedState.toString());
+
+        dumpPublicFile(FINISHED_FILE_PATH, finishedState.toString());
+    }
+
+    private static void dumpPublicFile(String path, String content) throws IOException {
+        File outputFile = getOutputFile(path);
+        try (OutputStream fileOutputStream = new FileOutputStream(outputFile, false)) {
+            fileOutputStream.write(content.getBytes());
+            System.out.println("File: " + outputFile.getAbsolutePath() + " successfully saved");
+        }
+
+        if (!outputFile.setReadable(true, false)) {
+            System.err.println("WARNING: Failed to make file " + outputFile.getAbsolutePath() + " readable!");
+        }
+    }
+
+    private static File getOutputFile(String path) {
+        return new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS), path);
```

Comment:
Why have you decided to use the `Environment.DIRECTORY_DOWNLOADS` folder instead of `Environment.DIRECTORY_DOCUMENTS`?

Reply:
Lets use documents, honestly I don't see any significant difference here